### PR TITLE
Feat: balancer group

### PR DESCRIPTION
**Pull request by Meldiron**
Add balancer group to proxy.
Cold-started executors are now preferred over round robin.

### CODE REVIEW

File: app/http.php [hunk 1]
```diff
@@ -83,21 +84,67 @@
...
+    $balancer1->addFilter(function ($option) {
+        /**
+         * @var array<string,mixed> $state
+         */
+        $state = \json_decode($option->getState('state', '{}'), true);
+        return ($state['usage'] ?? 0) < 80;
+    });
+
+    // Only low runtime-cpu usage
+    if ($runtimeId !== '') {
```
Comment (christyjacob4) [inline app/http.php:109]: I think its better to use !empty here

File: app/http.php [hunk 2]
```diff
@@ -83,21 +84,67 @@
...
+
+            return ($runtime['usage'] ?? 0) < 80;
+        });
+    }
+
+    // Any options
+    $balancer2 = new Balancer($algorithm);
+    $balancer2->addFilter(fn ($option) => $option->getState('status', 'offline') === 'online');
 
     foreach ($state as $state) {
```
Comment (christyjacob4) [inline app/http.php:134]: reminder to change the name of the loop variable

File: composer.json [hunk 3]
```diff
@@ -19,6 +19,12 @@
         "check": "./vendor/bin/phpstan analyse --level max -c phpstan.neon app src tests",
         "test": "./vendor/bin/phpunit --configuration phpunit.xml --debug"
     },
+    "repositories": [
+        {
+            "url": "https://github.com/utopia-php/balancer.git",
+            "type": "git"
+        }
+    ],
```
Comment (christyjacob4) [inline composer.json:27]: eldad has updated the repo on packagist. You can remove this section now

**Review by christyjacob4 — COMMENTED** [review]
Looks good . Lets continue 👍

File: app/http.php (hunk 2 above)
Comment (Meldiron) [reply app/http.php:134]: Changed in diffetent PR: https://github.com/open-runtimes/proxy/pull/7

**Review by christyjacob4 — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]

File: app/http.php [hunk 4]
```diff
@@ -83,21 +84,67 @@
 App::setResource('logger', fn () => $register->get('logger'));
 App::setResource('algorithm', fn () => $register->get('algorithm'));
 
-// Balancing must NOT be registry. This has to run on every request
-App::setResource('balancing', function (Table $state, Algorithm $algorithm) {
-    $balancing = new Balancing($algorithm);
+// Balancer must NOT be registry. This has to run on every request
+App::setResource('balancerGroup', function (Table $state, Algorithm $algorithm, Request $request) {
```
Comment (eldadfux) [inline app/http.php:88]: We might be able to just keep it simpler with naming it just `balancer`. This is our (only) balancer after all.

**Pull request merged**